tss-esapi/src/utils/credential.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2025 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::{
+    marker::PhantomData,
+    ops::{Add, Mul},
+};
+
+use cfb_mode::cipher::{AsyncStreamCipher, BlockCipherEncrypt};
+use digest::{
+    array::ArraySize,
+    consts::{B1, U8},
+    crypto_common::{Iv, KeyIvInit, KeySizeUser, WeakKeyError},
+    typenum::{
+        operator_aliases::{Add1, Sum},
+        Unsigned,
+    },
+    Digest, DynDigest, FixedOutputReset, Key, KeyInit, Mac, OutputSizeUser,
+};
+use ecdsa::elliptic_curve::{
+    ecdh::{EphemeralSecret, SharedSecret},
+    sec1::{Coordinates, FromEncodedPoint, ModulusSize, ToEncodedPoint},
+    AffinePoint, Curve, CurveArithmetic, FieldBytesSize, PublicKey,
+};
+use hmac::{EagerHash, Hmac};
+use log::error;
+use rand::{thread_rng, Rng};
+use rsa::{Oaep, RsaPublicKey};
+
+use crate::{
+    error::{Error, Result, WrapperErrorKind},
+    structures::{EncryptedSecret, IdObject, Name},
+    utils::kdf::{self},
+};
+
+type WeakResult<T> = core::result::Result<T, WeakKeyError>;
+
+// [`TpmHmac`] intends to code for the key expected for hmac
+// in the KDFa and KDFe derivations. There are no standard sizes for hmac keys really,
+// upstream RustCrypto considers it to be [BlockSize], but TPM specification
+// has a different opinion on the matter, and expect the key to the output
+// bit size of the hash algorithm used.
+//
+// See https://trustedcomputinggroup.org/wp-content/uploads/TPM-2.0-1.83-Part-1-Architecture.pdf#page=202
+// section 24.5 HMAC:
+//   bits the number of bits in the digest produced by ekNameAlg
+//
+// [BlockSize]: https://docs.rs/hmac/0.12.1/hmac/struct.HmacCore.html#impl-KeySizeUser-for-HmacCore%3CD%3E
+struct TpmHmac<H>(PhantomData<H>);
+
+impl<H> KeySizeUser for TpmHmac<H>
+where
+    H: OutputSizeUser,
+{
+    type KeySize = H::OutputSize;
+}
+
+pub fn make_credential_ecc<C, EkHash, EkCipher>(
```

Comment:
Can we get some documentation about the input parameters to this function please?